src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,435 @@
+package org.broadinstitute.hellbender.cmdline.GATKPlugin;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.reflections.ReflectionUtils;
+
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A plugin descriptor for managing the dynamic discovery of both @{InfoFieldAnnotation} and @{GenotypeAnnotation} objects
+ * within the packages defined the method getPackageNames() (default @{org.broadinstitute.hellbender.tools.walkers.annotator}).
+ * Also handles integrating annotation specific arguments from the command line with tool specified defaults.
+ *
+ * Unlike @{GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guaranteed to be
+ * preserved in all cases, especially when group annotations are involved.
+ */
+public class GATKAnnotationPluginDescriptor  extends CommandLinePluginDescriptor<Annotation> {
+    //TODO this should be a configurable option or otherwise exposed to the user when configurations are fully supported.
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.tools.walkers.annotator";
+    private static final Class<?> pluginBaseClass = org.broadinstitute.hellbender.tools.walkers.annotator.Annotation.class;
+
+    protected transient Logger logger = LogManager.getLogger(this.getClass());
+
+    @ArgumentCollection
+    private final GATKAnnotationArgumentCollection userArgs;
+
+    // Map of Annotation (simple) class names to the corresponding discovered plugin instance
+    private final Map<String, Annotation> allDiscoveredAnnotations = new HashMap<>();
+
+    // Map of Annotation (simple) class names to the corresponding default plugin instance
+    // it is a LinkedHashMap because we want to remember the order in which these were provided, and also keep the
```

Comment:
Do we actually care about the order (this comment seems at odds with the class level comment about order). f we don't then LinkHashMap is probably overkill.

Reply:
done